libs/content/workspace-ffi/src/apple/macos/api.rs

```diff
@@ -27,15 +27,15 @@ pub unsafe extern "C" fn key_event(
 
     let Some(key) = NSKeys::from(key_code) else { return };
 
-    // Event::Text
-    let text_modifiers = modifiers.shift_only() || modifiers.is_none();
-    if pressed && text_modifiers && key.valid_text() {
-        let text = CStr::from_ptr(characters).to_str().unwrap().to_string();
-        obj.raw_input.events.push(Event::Text(text));
-    }
+    let alt_without_ctrl_cmd =
+        modifiers.alt && !modifiers.ctrl && !modifiers.mac_cmd && !modifiers.command;
+    let text_modifiers = modifiers.shift_only() || alt_without_ctrl_cmd || modifiers.is_none();
+    let text = CStr::from_ptr(characters).to_str().unwrap().to_string();
+    let is_valid_text = text.chars().any(|c| !c.is_control());
```

Comment:
`.to_string()` makes a copy that's unnecessary bc you're just iterating to check for control chars